src/pardibaal/Federation.cpp

```diff
@@ -145,81 +145,159 @@ namespace pardibaal {
         return true;
     }
 
+    template<bool is_exact>
     relation_t Federation::relation(const DBM& dbm) const {
+        if (is_exact)
+            return this->relation<true>(Federation(dbm));
+        
         if (this->is_empty())
             return dbm.is_empty() ? relation_t::equal() : relation_t::subset();
         if (dbm.is_empty())
             return relation_t::superset();
+        if (this->dimension() != dbm.dimension())
+            return relation_t::different();
 
-        bool eq = false, sub = true;
+        bool diff = false, eq = false;
 
         for (const auto& e : zones) {
             auto relation = e.relation(dbm);
-            if (relation._superset && not relation._equal) return relation;
+            if (relation.is_superset()) return relation;
 
-            eq = eq || relation._equal;
-            sub = sub && relation._subset;
+            diff = diff || relation.is_different();
+            eq = eq || relation.is_equal();
         }
 
-        if (eq && sub) return relation_t::equal();      // If one dbm is equal and all others are equal/subsets
-        if (eq && !sub) return relation_t::superset();  // If one dbm is equal, but not all the dbm are equal/subsets
-        if (sub) return relation_t::subset();           // If no dbm is equal but all of them are subsets
+        // At this point: no dbms in this fed are supersets of dbm
 
-        return relation_t::different();
+        if (diff && eq) return relation_t::superset(); // If some dbms are equal and some are different
+
+        if (diff && not eq) return relation_t::different(); // If none are equal, but some are different.
+
+        if (eq) return relation_t::equal(); // If one dbm is equal and none are different/supersets
+        
+        return relation_t::subset(); // If no dbm is equal/different/supersets, then all must be subsets
     }
 
+    template<bool is_exact>
     relation_t Federation::relation(const Federation& fed) const {
         if (this->is_empty())
             return fed.is_empty() ? relation_t::equal() : relation_t::subset();
         if (fed.is_empty())
             return relation_t::superset();
+        if (this->dimension() != fed.dimension())
+            return relation_t::different();
+
+        if (is_exact) {
```

Comment:
if constexpr(is_exact) {

Reply:
added constexpr now to both places where `is_exact` is checked in templates